modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/filter/StructuredContentEdmProvider.java

```diff
@@ -0,0 +1,115 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.structured.content.apio.internal.architect.filter;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.olingo.commons.api.edm.EdmPrimitiveTypeKind;
+import org.apache.olingo.commons.api.edm.FullQualifiedName;
+import org.apache.olingo.commons.api.edm.provider.CsdlEntityContainer;
+import org.apache.olingo.commons.api.edm.provider.CsdlEntitySet;
+import org.apache.olingo.commons.api.edm.provider.CsdlEntityType;
+import org.apache.olingo.commons.api.edm.provider.CsdlProperty;
+import org.apache.olingo.commons.api.edm.provider.CsdlSchema;
+
+/**
+ * Provides the entity data model from the Indexed Entity (JournalArticle).
+ *
+ * @author Julio Camarero
+ * @review
+ */
+public class StructuredContentEdmProvider
+	extends SingleEntitySchemaBaseProvider {
+
+	public StructuredContentEdmProvider() {
+		addSchema(_createCsdlSchema());
+	}
+
+	@Override
+	public String getSingleEntityTypeName() {
+		return _ENTITY_NAME;
+	}
+
+	private CsdlEntityContainer _createCsdlEntityContainer() {
+		CsdlEntityContainer csdlEntityContainer = new CsdlEntityContainer();
+
+		CsdlEntityType csdlEntityType = _createCsdlEntityType();
+
+		FullQualifiedName fullQualifiedName = new FullQualifiedName(
+			_NAMESPACE, csdlEntityType.getName());
+
+		CsdlEntitySet csdlEntitySet = new CsdlEntitySet();
+
+		csdlEntitySet.setName(csdlEntityType.getName());
+		csdlEntitySet.setType(fullQualifiedName);
+
+		csdlEntityContainer.setName(_ENTITY_NAME);
+		csdlEntityContainer.setEntitySets(
+			Collections.singletonList(csdlEntitySet));
+
+		return csdlEntityContainer;
+	}
+
+	private CsdlEntityType _createCsdlEntityType() {
+		CsdlProperty titleCsdlProperty = _createCsdlProperty(
+			_TITLE_FIELD, EdmPrimitiveTypeKind.String.getFullQualifiedName());
+
+		CsdlEntityType csdlEntityType = new CsdlEntityType();
+
+		csdlEntityType.setName(_ENTITY_NAME);
+
+		List<CsdlProperty> csdlPropertyList = new ArrayList<>();
+
+		csdlPropertyList.add(titleCsdlProperty);
+
+		csdlEntityType.setProperties(csdlPropertyList);
+
+		return csdlEntityType;
+	}
+
+	private CsdlProperty _createCsdlProperty(
+		String name, FullQualifiedName fullQualifiedName) {
+
+		CsdlProperty csdlProperty = new CsdlProperty();
+
+		csdlProperty.setName(name);
+
+		csdlProperty.setType(fullQualifiedName);
+
+		return csdlProperty;
+	}
+
+	private CsdlSchema _createCsdlSchema() {
+		CsdlSchema csdlSchema = new CsdlSchema();
+
+		csdlSchema.setNamespace(_NAMESPACE);
+
+		csdlSchema.setEntityTypes(
+			Collections.singletonList(_createCsdlEntityType()));
+
+		csdlSchema.setEntityContainer(_createCsdlEntityContainer());
+
+		return csdlSchema;
+	}
+
+	private static final String _ENTITY_NAME = "StructuredContent";
+
+	private static final String _NAMESPACE = "HeadlessApio";
```

Comment:
does the namespace have to be in upper camel case? I've seen string constants in lower case and separated by _

Reply:
But if they are public, it's weird. Better leave it that way if they are part of the API.